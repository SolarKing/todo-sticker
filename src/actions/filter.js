// @flow

export type FilterActions =
    | DueDateFilterAction;

export type DueDateFilterAction = {|
    type: 'dueDateFilter',
    filter: string,
|}

export function settingDueDateFilter(dueDateFilter: string): DueDateFilterAction {
    return {
        type: 'dueDateFilter',
        filter: dueDateFilter
    };
}
