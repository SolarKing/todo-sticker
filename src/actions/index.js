// @flow
import type { TodoActions } from './todo';
import type { FilterActions } from './filter';
import type { SortActions } from './sort';

export type Actions =
    | TodoActions
    | FilterActions
    | SortActions;
