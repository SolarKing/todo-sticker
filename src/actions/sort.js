// @flow
import type {SortByType, SortColumn} from '../constant/sort';

export type SortActions =
    | SortByAction
    | SortColumnAction;

export type SortColumnAction = {|
    type: 'sortColumn',
    column: SortColumn
|};

export type SortByAction = {|
    type: 'sortBy',
    sortBy: SortByType
|};

export function settingSortColumn(column: SortColumn): SortColumnAction {
    return {
        type: 'sortColumn',
        column: column
    };
}
