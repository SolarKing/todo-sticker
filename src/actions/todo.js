// @flow
import type { TodoState } from '../states';

export type TodoActions = AddAction
    | EditAction
    | DeleteAction
    | CompleteAction;

export const TodoActionTypeEnum = {
    Add: 'add',
    Edit: 'edit',
    Delete: 'delete',
    Complete: 'complete'
};

export type AddAction = {|
    type: 'add',
    todoState: TodoState
|}

export type EditAction = {|
    type: 'edit',
    todoState: TodoState
|}

export type DeleteAction = {|
    type: 'delete',
    id: string
|}

export type CompleteAction = {|
    type: 'complete',
    id: string
|}

export function addTodo(todoState: TodoState): AddAction {
    return {
        type: TodoActionTypeEnum.Add,
        todoState
    };
}

export function editTodo(todoState: TodoState): EditAction {
    return {
        type: TodoActionTypeEnum.Edit,
        todoState
    };
}

export function deleteTodo(id: string): DeleteAction {
    return {
        type: TodoActionTypeEnum.Delete,
        id
    };
}

export function completeTodo(id: string): CompleteAction {
    return {
        type: TodoActionTypeEnum.Complete,
        id
    };
}
