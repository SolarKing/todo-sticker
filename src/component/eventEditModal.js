// @flow
import { Modal, DatePicker, TimePicker, Input } from 'antd';
import React from 'react';
import ReactDOM from 'react-dom';
import moment from 'moment';
import Translate from '../class/translate';
import type {TodoState} from '../states';

type Props = {
    title: string,
    todoState?: TodoState,
    onSave: (newTodoState: TodoState) => void,
}

type States = {
    todoState: TodoState,
    visible: boolean,
}

export default class EditModal extends React.Component<Props, States> {

    static defaultProps = {
        visible: false,
        defaultDescriptin: '',
    }

    constructor(props: Props) {
        super(props);

        let defaultDate = new Date();
        defaultDate.setUTCSeconds(0);

        const {
            todoState = {
                dueDatetime: defaultDate.getTime(),
                description: ''
            }
        } = this.props;

        this.state = {
            todoState,
            visible: true,
        };
    }

    _handleDateChanged(dates: moment) {
        let {
            todoState
        } = this.state;

        let dueDatetime = moment(todoState.dueDatetime);
        dueDatetime.set('year', dates.get('year'));
        dueDatetime.set('month', dates.get('month'));
        dueDatetime.set('date', dates.get('date'));
        todoState.dueDatetime = dueDatetime.toDate().getTime();

        this.setState({
            todoState
        });
    }

    _handleTimeChanged(time: moment) {
        let {
            todoState
        } = this.state;

        let dueDatetime = moment(todoState.dueDatetime);
        dueDatetime.set('hour', time.get('hour'));
        dueDatetime.set('minute', time.get('minute'));
        todoState.dueDatetime = dueDatetime.toDate().getTime();

        this.setState({
            todoState
        });
    }

    _handleDescriptionChanged(event: SyntheticEvent<HTMLTextAreaElement>) {
        let {
            todoState
        } = this.state;
        todoState.description = event.currentTarget.value;

        this.setState({
            todoState
        });
    }

    _handleOk() {
        this.props.onSave(this.state.todoState);
        this.setState({
            visible: false,
        });
    }

    _handleCancel() {
        this.setState({
            visible: false,
        });
    }

    render() {
        const {
            onSave,
            title,
            // onClose,
            ...others
        } = this.props;

        const {
            todoState
        } = this.state;

        return (
            <Modal
                {...others}
                title={title}
                destroyOnClose={true}
                onOk={this._handleOk.bind(this)}
                onCancel={this._handleCancel.bind(this)}
                visible={this.state.visible}
            >
                <p>{Translate.tr('Due date')}</p>
                <DatePicker 
                    allowClear={false}
                    onChange={this._handleDateChanged.bind(this)}
                    defaultValue={moment(todoState.dueDatetime)}
                />
                &nbsp;
                <TimePicker
                    allowEmpty={false}
                    onChange={this._handleTimeChanged.bind(this)}
                    defaultValue={moment(todoState.dueDatetime)}
                    format={'HH:mm'}
                />
                <p>{Translate.tr('eventDescription')}</p>
                <Input.TextArea
                    defaultValue={todoState.description}
                    onChange={this._handleDescriptionChanged.bind(this)}
                    rows={4}
                />
            </Modal>
        );
    }
}

export const TodoEditableModal = {
    show(props: Props) {
        let wrapper = document.getElementById('todoModal');
        if (!wrapper) {
            return;
        }

        let cleanup = () => {
            ReactDOM.unmountComponentAtNode(wrapper);
        };

        ReactDOM.render(
            <EditModal
                {...props}
                afterClose={cleanup}
            />
            , wrapper
        );
    }
};