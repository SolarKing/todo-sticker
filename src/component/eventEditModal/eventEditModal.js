// @flow
import { Modal, DatePicker, TimePicker, Input } from 'antd';
import React from 'react';
import moment from 'moment';
import 'moment/locale/zh-cn';
import Translate from '../../class/translate';

export type EditInfo = {
    date: number,
    time: number,
    desctiption: string
}

type Props = {
    visible: boolean,
    title: string,
    defaultDate?: number,
    defaultTime?: number,
    defaultDescriptin?: string,
    onSave?: (editInfo: EditInfo) => void
}

type States = {
    date: number,
    time: number,
    desctiption: string
}

export default class EditModal extends React.Component<Props, States> {

    handleDateChanged: Function;
    handleTimeChanged: Function;
    handleDescriptionChanged: Function;
    handleOk: Function;

    static defaultProps = {
        visible: false,
        defaultDate: new Date().getTime(),
        defaultTime: new Date().getTime(),
        defaultDescriptin: '',
    }

    state = {
        date: new Date().getTime(),
        time: new Date().getTime(),
        desctiption: ''
    };

    constructor(props: Props) {
        super(props);
        moment.locale('en');
        this.handleDateChanged = this.handleDateChanged.bind(this);
        this.handleTimeChanged = this.handleTimeChanged.bind(this);
        this.handleDescriptionChanged = this.handleDescriptionChanged.bind(this);
        this.handleOk = this.handleOk.bind(this);
    }

    handleDateChanged(dates: moment) {
        this.setState(
            {
                date: dates
            }
        );
    }

    handleTimeChanged(time: moment) {
        this.setState(
            {
                time: time
            }
        );
    }

    handleDescriptionChanged(event: SyntheticEvent<HTMLTextAreaElement>) {
        let descriptin = event.currentTarget.value;
        this.setState({
            desctiption: descriptin
        });
    }

    handleOk() {
        if (!this.props.onSave) {
            return;
        }
        let editInfo: EditInfo;
        editInfo = {
            date: this.state.date,
            time: this.state.time,
            desctiption: this.state.desctiption
        };
        this.props.onSave(editInfo);
    }

    render() {
        var {
            defaultDate,
            defaultTime,
            defaultDescriptin,
            ...others
        } = this.props;

        return (
            <Modal onOk={this.handleOk} {...others} >
                <p>{Translate.tr('notificationDate')}</p>
                <DatePicker onChange={this.handleDateChanged} value={moment(defaultDate)} />
                <p>{Translate.tr('notificationTime')}</p>
                <TimePicker onChange={this.handleTimeChanged} value={moment(defaultTime)} format={'HH:mm'} />
                <p>{Translate.tr('eventDescription')}</p>
                <Input.TextArea defaultValue={defaultDescriptin} onChange={this.handleDescriptionChanged} rows={4} />
            </Modal>
        );
    }
}
