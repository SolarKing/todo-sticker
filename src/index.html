<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="UTF-8">
	<title>Todo Sticker</title>
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
</head>
<body>
	<div id="todoModal"></div>
	<div id="app"></div>
</body>
</html>
