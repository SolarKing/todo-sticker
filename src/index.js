// @flow

import React from 'react';
import ReactDOM from 'react-dom';
import TodosPage from './component/todosPage';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from './reducers';
import '../node_modules/antd/dist/antd.min.css';
import '../node_modules/antd/dist/antd.min.js';
import type {TodoState} from './states/index';
import NotifyServer from './class/notify/notifyServer';

// localStorage.removeItem('todos');
const initStorageData = localStorage.getItem('todos') || '{}';
const store = createStore(rootReducer, JSON.parse(initStorageData));
store.subscribe(() => {
    localStorage.setItem('todos', JSON.stringify(store.getState()));
    console.log(store.getState());
});

let todayTimestamp: number = new Date().getTime();

store.getState().todos.forEach((todo: TodoState) => {
    if (todo.complete) {
        return;
    }
    if (todo.dueDatetime <= todayTimestamp) {
        return;
    }
    NotifyServer.instance.addSchedule(todo.id || 'empty', todo.dueDatetime, todo.description);
});

const app = document.getElementById('app');

app && ReactDOM.render(
    <Provider store={store}>
        <TodosPage />
    </Provider>,
    app
);
