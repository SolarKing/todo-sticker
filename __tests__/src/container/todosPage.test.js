// @flow
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import TodosPage from '../../../src/container/todosPage';
import FilterPanelContainer from '../../../src/container/filterPanelContainer';
import { TodoEditableModal } from '../../../src/component/eventEditModal';
import configureStore from 'redux-mock-store';
import { addTodo } from '../../../src/actions';
import { Button } from 'antd';

describe('TodosPage', () => {
    let wrapperContainer: ShallowWrapper;
    let wrapper: ShallowWrapper;
    let store = configureStore()();

    beforeAll(() => {
        wrapperContainer = shallow(
            <TodosPage 
                store={store}
            />
        );
        wrapper = wrapperContainer.shallow();
    });

    it('should render correct', () => {
        expect(wrapperContainer).toMatchSnapshot();
    });

    it('should render correct with origin component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should show filter panel', () => {
        wrapper.find(Button).simulate('click');
        expect(wrapper.state().showPanel).toBeTruthy();
    });

    it('should hide filter panel', () => {
        wrapper.find(FilterPanelContainer).simulate('hidden');
        expect(wrapper.state().showPanel).toBeFalsy();
    });

    it('should show editable modal', () => {
        let show = jest.spyOn(TodoEditableModal, 'show');
        wrapper.find('AddButton').simulate('click');
        expect(show.mock.calls.length).toEqual(1);
    });

    it('should call addTodo', () => {
        let todoState = {
            dueDatetime: 0,
            description: 'Hello',
        };
        let show = jest.spyOn(TodoEditableModal, 'show');
        wrapper.find('AddButton').simulate('click');
        show.mock.calls[0][0].onSave(todoState);
        expect(store.getActions()).toEqual([addTodo(todoState)]);
    });

});
