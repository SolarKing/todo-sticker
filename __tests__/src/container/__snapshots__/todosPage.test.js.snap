// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodosPage should render correct 1`] = `
<TodosPage
  addTodo={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`TodosPage should render correct with origin component 1`] = `
<todosPage__TodosContainer>
  <todosPage__ToolBar>
    <AddButton
      onClick={[Function]}
    >
      <span
        className="glyphicon glyphicon-plus-sign"
      />
    </AddButton>
  </todosPage__ToolBar>
  <todosPage__TodosView>
    <todosPage__FilterButtonDiv>
      <FilterButton
        onClick={[Function]}
      />
      <Connect(FilterPanel)
        onHidden={[Function]}
        show={false}
      />
    </todosPage__FilterButtonDiv>
    <todosPage__TodoListDiv>
      <Connect(TodoList) />
    </todosPage__TodoListDiv>
    <todosPage__SortButtonDiv
      style={
        Object {
          "display": "inline",
        }
      }
    />
  </todosPage__TodosView>
</todosPage__TodosContainer>
`;
